snuba/clusters/storage_sets.py

```diff
@@ -0,0 +1,30 @@
+from typing import Mapping, Set
+
+from enum import Enum
+from snuba.datasets.storages import StorageKey
+
+
+class StorageSetKey(Enum):
+    EVENTS = "events"
+    GROUPASSIGNEES = "groupassignees"
+    OUTCOMES = "outcomes"
+    QUERYLOG = "querylog"
+    SESSIONS = "sessions"
+    TRANSACTIONS = "transactions"
+
+
+STORAGE_SETS: Mapping[StorageSetKey, Set[StorageKey]] = {
+    StorageSetKey.EVENTS: {
+        StorageKey.ERRORS,
+        StorageKey.EVENTS,
+        StorageKey.GROUPEDMESSAGES,
+        # TODO: Remove once groups are no longer storages
+        StorageKey.GROUPS,
+    },
+    # Should group assignee be on the events cluster?
+    StorageSetKey.GROUPASSIGNEES: {StorageKey.GROUPASSIGNEES},
+    StorageSetKey.OUTCOMES: {StorageKey.OUTCOMES_RAW, StorageKey.OUTCOMES_HOURLY},
+    StorageSetKey.QUERYLOG: {StorageKey.QUERYLOG},
+    StorageSetKey.SESSIONS: {StorageKey.SESSIONS_RAW, StorageKey.SESSIONS_HOURLY},
+    StorageSetKey.TRANSACTIONS: {StorageKey.TRANSACTIONS},
```

Comment:
It seems this relationship is going the wrong way.
It is still a mapping between StorageSet and a set of Storages.
This means it could be an 1:n or an m:n.
I think you should have a get_storage_set method in the storage. That forces a 1:n relationship between storageset and storage. Like we do for foreign keys in databases.

Then getting the cluster from storage becomes trivial and you may not even need the storageKey anymore.